vacasa/connect/util.py

```diff
@@ -10,15 +10,24 @@
 
 def log_http_error(r: Response):
     """ Log any 4XX/5XX error responses and re-raise """
-    try:
-        r.raise_for_status()
-    except RequestException as e:
+    if r.status_code < 500:
         try:
-            logger.exception(r.json())
-        except ValueError:
-            logger.exception(r.content)
-        raise e
-
+            r.raise_for_status()
+        except RequestException as e:
+            try:
+                logger.info(r.json())
+            except ValueError:
+                logger.info(r.content)
+            raise e
+    else:
+        try:
+            r.raise_for_status()
+        except RequestException as e:
+            try:
+                logger.exception(r.json())
+            except ValueError:
+                logger.exception(r.content)
+            raise e
```

Comment:
I'm pretty sure if you do `logger.log()` you can pass in the log level dynamically which could simplify this code a lot. It would look more like this:
```python
def log_http_error(r: Response):
    """ Log any 4XX/5XX error responses and re-raise """
    try:
        r.raise_for_status()
    except RequestException as e:
        log_level = 20 if r.status_code < 500 else 40
        try:
            logger.log(log_level, r.json())
        except ValueError:
            logger.log(log_level, r.content)
        raise e
```

Reply:
Much better, I didn't really like the way the code looked

Reply:
Cool. I did that from memory so you'll have to make sure it does what I think it does. Also, as Bijan pointed out to me, it would be more readable to lookup the log level values by name instead of just using the integers like I did.